comms/core/src/protocol/messaging/error.rs

```diff
@@ -68,6 +68,10 @@ impl From<io::Error> for MessagingProtocolError {
             ErrorKind::ConnectionReset |
             ErrorKind::ConnectionAborted |
             ErrorKind::BrokenPipe |
+            ErrorKind::WriteZero |
+            ErrorKind::NotConnected |
+            ErrorKind::AddrInUse |
+            ErrorKind::AddrNotAvailable |
```

Comment:
I would make  `ErrorKind::AddrInUse` and `ErrorKind::AddrNotAvailable` errors (this socket is a substream so these would not be expected at all and if they occur, we want to know). We only want to catch disconnection errors that can occur in normal operation. NotConnected seems like a good addition.

Reply:
Thanks for the explanation @sdbondi.
It's done.